### DISCUSSION TITLE
Is diesel strict encapsulated?

### DISCUSSION THREAD

**Discussion by siqueira-prog** [diesel-rs/diesel#4380] [Q&A]
I'm asking this because i was doing some researchs. Then i discover that rust uses the structs to create the logic behinds. But theres no private and protect concepts... Is it? (Sorry if i ask something wrong, i'm beguinner...) I imagined that could be something about the language for itself that do this and theres no need to add this concepts... Is it?

**Comment by weiznich** [comment]
I'm not sure what you exactly trying to ask here. Maybe you can add some more details to your question to get a better answer.

If that refers to accessing diesel internals: No that's generally not possible from outside diesel, as they are private based on rusts module privacy.
